### PR TITLE
Remove extra call to Bundler for Sponsored User ops

### PR DESCRIPTION
**Pull request by livingrockrises**
# Summary

using optimised flow for gasless, biconomy paymaster 

older code ref getGasFeeValues here

https://github.com/bcnmy/biconomy-client-sdk/blob/r7/packages/account/src/BiconomySmartAccountV2.ts#L468 

Related Issue: # (issue number)

## Change Type

- [ ] Bug Fix
- [ ] Refactor
- [ ] New Feature
- [ ] Breaking Change
- [ ] Documentation Update
- [ ] Performance Improvement
- [ ] Other

# Checklist

- [ ] My code follows this project's style guidelines
- [ ] I've reviewed my own code
- [ ] I've added comments for any hard-to-understand areas
- [ ] I've updated the documentation if necessary
- [ ] My changes generate no new warnings
- [ ] I've added tests that prove my fix is effective or my feature works
- [ ] All unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published

# Additional Information

# Branch Naming

### CODE REVIEW

File: packages/account/src/BiconomySmartAccountV2.ts [hunk 1]
```diff
@@ -64,7 +64,7 @@ import { BiconomyFactoryAbi } from "./abi/Factory.js";
 import { BiconomyAccountAbi } from "./abi/SmartAccount.js";
 import { AccountResolverAbi } from "./abi/AccountResolver.js";
 import { Logger } from "@biconomy/common";
-import { FeeQuotesOrDataDto, FeeQuotesOrDataResponse } from "@biconomy/paymaster";
+import { /*BiconomyPaymaster,*/ BiconomyPaymaster, FeeQuotesOrDataDto, FeeQuotesOrDataResponse } from "@biconomy/paymaster";
```
Comment (joepegler) [inline packages/account/src/BiconomySmartAccountV2.ts:67]: remove

**Review by joepegler — APPROVED** [review]
LGTM

**Comment by joepegler** [comment]
Conflicts

**Review by VGabriel45 — APPROVED** [review]

File: packages/account/src/BiconomySmartAccountV2.ts [hunk 2]
```diff
@@ -1043,16 +1043,27 @@ export class BiconomySmartAccountV2 extends BaseSmartContractAccount {
...
-    if (buildUseropDto?.paymasterServiceData) {
-      userOp = await this.getPaymasterUserOp(userOp, buildUseropDto.paymasterServiceData);
-    }
+      // populate gasfee values and make a call to paymaster
+      userOp.maxFeePerGas = gasFeeValues?.maxFeePerGas as Hex;
+      userOp.maxPriorityFeePerGas = gasFeeValues?.maxPriorityFeePerGas as Hex;
 
-    Logger.log("UserOp after estimation ", userOp);
+      userOp = await this.getPaymasterUserOp(userOp, buildUseropDto.paymasterServiceData);
+      return userOp;
```
Comment (AmanRaj1608) [inline packages/account/src/BiconomySmartAccountV2.ts:1058]: Should test the cases
- Developer pass the mode as sponsored but didn't added it on dashboard.
- Added the address but haven't whitelisted the method.

As there is no try catch what is it returning, half user op or error.

**Review by AmanRaj1608 — APPROVED** [review]

File: packages/account/src/BiconomySmartAccountV2.ts (hunk 2 above)
Comment (VGabriel45) [reply packages/account/src/BiconomySmartAccountV2.ts:1058]: Right, will add a try catch and throw and error

Comment (VGabriel45) [reply packages/account/src/BiconomySmartAccountV2.ts:1058]: @AmanRaj1608 actually, this code will throw if method is not whitelisted or if anything else is wrong with paymaster. No need for a try catch. Added a test which checks that the sendTransaction throws an error if a method is not whitelisted.

**Pull request merged**